src/compiler/checker.ts

```diff
@@ -11143,12 +11144,13 @@ namespace ts {
                     someHaveQuestionToken = someHaveQuestionToken || hasQuestionToken(node);
                     allHaveQuestionToken = allHaveQuestionToken && hasQuestionToken(node);
 
-                    if (nodeIsPresent(node.body) && bodyDeclaration) {
+                    if (nodeIsPresent(node.body) && implementationDeclaration) {
                         if (isConstructor) {
                             multipleConstructorImplementation = true;
                         }
                         else {
                             duplicateFunctionDeclaration = true;
+                            duplicateDefaultExports = !!(node.flags & NodeFlags.Default);

```

Comment:
Actually, I realized that we don't need the extra variable, we can just check the implementation.